test_conformance/vulkan/test_vulkan_interop_buffer.cpp

```diff
@@ -229,14 +239,39 @@ int run_test_with_two_queue(cl_context &context, cl_command_queue &cmd_queue1,
 
                 if (iter == 0)
```

Comment:
This code can be simplified.  We do not need to wait on a fence from the previous frame, so there is no need to check iter when using fence.  I suggest we always reset the fence at the start of the loop if use_fence.  Something like

`
if (use_fence)
{
    fence->reset();
    vkQueue.submit(vkCommandBuffer, fence);
}
else
{
    if (iter == 0)
    {
        vkQueue.submit(vkCommandBuffer, vkVk2CLSemaphore);
    }
    else
    {
        vkQueue.submit(vkCl2VkSemaphore, vkCommandBuffer,
                       vkVk2CLSemaphore);
    }
}
`